atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/factory/InMemoryLockAndTimestampServiceFactoryV2.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory;
+
+import com.palantir.timelock.paxos.InMemoryTimelockExtension;
+
+public class InMemoryLockAndTimestampServiceFactoryV2 implements LockAndTimestampServiceFactory {
+    private final InMemoryTimelockExtension inMemoryTimelockExtension;
+
+    public InMemoryLockAndTimestampServiceFactoryV2(InMemoryTimelockExtension inMemoryTimelockExtension) {
+        this.inMemoryTimelockExtension = inMemoryTimelockExtension;
+    }
```

Comment:
Exact replica of `InMemoryLockAndTimestampServiceFactory` except here it uses `InMemoryTimelockExtension` instead of `InMemoryTimelockServices`.